Cython/Utility/TypeConversion.c

```diff
@@ -276,30 +276,87 @@ static CYTHON_INLINE const char* __Pyx_PyUnicode_AsStringAndSize(PyObject* o, Py
 static CYTHON_INLINE const char* __Pyx_PyObject_AsStringAndSize(PyObject* o, Py_ssize_t *length) {
 #if __PYX_DEFAULT_STRING_ENCODING_IS_ASCII || __PYX_DEFAULT_STRING_ENCODING_IS_UTF8
     if (PyUnicode_Check(o)) {
-        return __Pyx_PyUnicode_AsStringAndSize(o, length);
+        char* result = __Pyx_PyUnicode_AsStringAndSize(o, length);
+        if (unlikely(result == NULL)) {
+            goto error;
+        } else {
+            return result;
+        }
     } else
 #endif
 
     if (PyByteArray_Check(o)) {
+        int r = PyErr_Warn(
+            PyExc_RuntimeWarning,
+            "Cython: Pointer taken from resizable `bytearray` object. "
+            "Consider using `unsigned char[::1]` instead. "
+            "This will also ensure the underlying `bytearray` is not resized while held."
+        );
+        if (unlikely(r < 0)) {
+            return NULL;
+        } else {
+            return result;
+        }
```

Comment:
Tried to add a warning in relation to issue: https://github.com/cython/cython/issues/6760

Since we are providing an alternative path, which would avoid that bug here, it makes sense to encourage its use

Please let me know what you think